src/com/amazon/ion/util/Equivalence.java

```diff
@@ -173,6 +175,7 @@ public static final class Builder {
 
         private boolean isStrict = true;
         private Double epsilon = null;
+        private int maxComparisonDepth = 1000;
```

Comment:
Of course there is error on both sides; there are both JVM setups for which this is too high and _theoretical_ datasets for which it is too low. I looked at this last night and slept on it.

FWIW the default recursion limit in cpython is 1000. So at the least is the approximately consistent with ion-python and reflects IMO a pretty decent default.